### DISCUSSION TITLE
Are the docs still correct in suggesting that `Annotated[..., Field(default=...)]` is unsupported?

### DISCUSSION THREAD

**Discussion by blakeNaccarato** [pydantic/pydantic#9379] [Question]
In the docs for [using `Annotated`](https://docs.pydantic.dev/dev/concepts/fields/#using-annotated) (both stable, latest, and dev), a note is given that "the `Field.default` argument is not supported inside `Annotated`". Is this still true, or should that warning be removed? Here are the relevant lines:

https://github.com/pydantic/pydantic/blob/4d7bef62aeff10985fe67d13477fe666b13ba070/docs/concepts/fields.md?plain=1#L57-L59

I ask because `flag: Annotated[bool, Field(default_factory=lambda: False)]` is unwieldy compared with `flag: Annotated[bool, Field(default=False)]`, for instance, and the latter works just fine for model instantiation. Possibly it breaks down in some other usage, though, so that might need clarification. Of course you could do `flag: bool = False`, but say I'm rolling a model with primarily `Annotated` fields, I might as well use `Annotated` everywhere, even in the case of default `bool`s, for instance.

Even the code example given in the docs works fine with a straight `default` instead of `default_factory`. Here it is now:

```Python
class User(BaseModel):
    id: Annotated[str, Field(default_factory=lambda: uuid4().hex)]
```

And here it is more simply stated:

```Python
class User(BaseModel):
    id: Annotated[str, Field(default=uuid4().hex)]
```

In fact, in this toy example it's harmful to use `default_factory` (to get around a possibly phantom limitation?), because it's actually a constant, evaluated once, so every user instantiated without an `id` will have the same `id` in a given session.

**EDIT:** Hmm, I guess if the expected interface to the user is to require that field (by type hints telling them it's missing), then specifying defaults in `Annotated` supports a default on the "backend" while still prompting the user for a value, but maybe it's just not a great practice to specify defaults inside of `Annotated` anyways.

**Comment by kc0506** [comment]
I am also confused by this note. I search the tests and find the use of `Field.default` in `Annotated` [here](https://github.com/pydantic/pydantic/blob/main/tests/test_json_schema.py#L5683) and [here](https://github.com/pydantic/pydantic/blob/main/tests/test_types.py#L6352). 
I guess the note is out-dated, but I am not sure.
